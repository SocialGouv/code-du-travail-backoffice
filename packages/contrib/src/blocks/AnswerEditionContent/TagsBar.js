import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import Tag from "../../elements/Tag";

const Container = styled(Flex)`
  background-color: #d7d7d7;
  color: #555555;
  padding: 0 1rem;
`;
const Title = styled.p`
  font-weight: 600;
  font-size: 0.8rem;
  margin: 1rem 0 0.75rem;
  text-transform: uppercase;
  user-select: none;
`;

export default class extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      selectedTags: this.props.selectedTags
    };
  }

  toggleTag(id) {
    if (this.state.selectedTags.includes(id)) {
      this.setState({
        selectedTags: this.state.selectedTags.filter(_id => _id !== id)
      });
      this.props.onToggleTag(id, false);

      return;
    }

    this.setState({ selectedTags: [...this.state.selectedTags, id] });
    this.props.onToggleTag(id, true);
  }

  getTags(category) {
    return this.props.tags
      .filter(tag => tag.category === category)
      .map(({ id, value }, index) => (
        <Tag
          key={index}
          onClick={() => this.toggleTag(id)}
          selected={this.state.selectedTags.includes(id)}
        >
          {value}
        </Tag>
      ));
  }

  render() {
    return (
      <Container className={this.props.className} flexDirection="column">
        <Flex flexDirection="column">
          <Title>Type de contrat</Title>
          <Flex flexWrap="wrap">{this.getTags("contract_type")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Title>Cible</Title>
          <Flex flexWrap="wrap">{this.getTags("target")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Title>Dur√©e de travail</Title>
          <Flex flexWrap="wrap">{this.getTags("work_time")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Title>{"Type d'horaires"}</Title>
          <Flex flexWrap="wrap">{this.getTags("work_schedule_type")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Title>Particularismes</Title>
          <Flex flexWrap="wrap">{this.getTags("distinctive_identity")}</Flex>
        </Flex>
      </Container>
    );
  }
}
