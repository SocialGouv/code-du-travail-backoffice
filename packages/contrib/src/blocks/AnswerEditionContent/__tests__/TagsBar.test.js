import React from "react";
import { fireEvent, render } from "react-testing-library";

import TagsBar from "../TagsBar";

describe("[Contrib] blocks/AnswerEditionContent/<TagsBar />", () => {
  const onToggleTag = jest.fn();

  const props = {
    onToggleTag,
    selectedTags: ["da68fa2d-4bfc-4b5b-bea3-f39f44719650"],
    tags: [
      {
        id: "da68fa2d-4bfc-4b5b-bea3-f39f44719650",
        value: "A Tag",
        category: "contract_type"
      },
      {
        id: "d7d54d79-b389-416c-9e06-43eb36a5ab30",
        value: "Another Tag",
        category: "target"
      }
    ]
  };

  const { container, getByText } = render(<TagsBar {...props} />);

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should trigger onToggleTag() with the expected param", () => {
    fireEvent.click(getByText(/Another Tag/));
    expect(onToggleTag).toHaveBeenCalledWith(props.tags[1].id, true);
  });

  it("should trigger onToggleTag() with the expected param", () => {
    fireEvent.click(getByText(/Another Tag/));
    expect(onToggleTag).toHaveBeenCalledWith(props.tags[1].id, false);
  });
});
