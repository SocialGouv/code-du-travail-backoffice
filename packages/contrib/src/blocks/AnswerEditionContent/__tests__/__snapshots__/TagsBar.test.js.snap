// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/AnswerEditionContent/<TagsBar /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH TagsBar__Container-sc-1lrdtxx-0 cEsfMC"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <p
        class="TagsBar__Title-sc-1lrdtxx-1 jrzJtQ"
      >
        Type de contrat
      </p>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      >
        <span
          class="Tag-sc-16selk6-0 gFNzpE"
        >
          A Tag
        </span>
      </div>
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <p
        class="TagsBar__Title-sc-1lrdtxx-1 jrzJtQ"
      >
        Cible
      </p>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      >
        <span
          class="Tag-sc-16selk6-0 vTDzf"
        >
          Another Tag
        </span>
      </div>
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <p
        class="TagsBar__Title-sc-1lrdtxx-1 jrzJtQ"
      >
        Dur√©e de travail
      </p>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <p
        class="TagsBar__Title-sc-1lrdtxx-1 jrzJtQ"
      >
        Type d'horaires
      </p>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <p
        class="TagsBar__Title-sc-1lrdtxx-1 jrzJtQ"
      >
        Particularismes
      </p>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
  </div>
</div>
`;
